pkg/group/service_test.go

```diff
@@ -0,0 +1,318 @@
+package group
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/dhis2-sre/im-user/internal/errdef"
+	"github.com/dhis2-sre/im-user/pkg/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"gorm.io/gorm"
+)
+
+func Test_service_GetClusterConfiguration_Happy(t *testing.T) {
+	id := uint(1)
+	groupName := "whatever"
+	clusterConfiguration := &model.ClusterConfiguration{
+		Model:                   gorm.Model{ID: id},
+		GroupName:               groupName,
+		KubernetesConfiguration: nil,
+	}
+
+	repository := &mockGroupRepository{}
+	repository.
+		On("getClusterConfiguration", groupName).
+		Return(clusterConfiguration, nil)
+	userService := &mockUserService{}
+
+	service := NewService(repository, userService)
+
+	cc, err := service.GetClusterConfiguration(groupName)
+	require.NoError(t, err)
+
+	assert.Equal(t, id, cc.ID)
+	assert.Equal(t, groupName, cc.GroupName)
+}
+
+func Test_service_AddClusterConfiguration_Happy(t *testing.T) {
+	clusterConfiguration := &model.ClusterConfiguration{}
+	repository := &mockGroupRepository{}
+	repository.On("addClusterConfiguration", clusterConfiguration).Return(nil)
+
+	userService := &mockUserService{}
+
+	service := NewService(repository, userService)
+
+	err := service.AddClusterConfiguration(clusterConfiguration)
+
+	require.NoError(t, err)
```

Comment:
is there anything else we should assert on? did you miss calling `AssertExpectations` on the mock? same for tests below